tests/rpm_tests/test_create_rpm.py

```diff
@@ -453,3 +475,47 @@ def apply_services_prefix(path):
     @staticmethod
     def apply_dbus_prefix(path):
         return ModifyingFilters.apply_rpm_prefix("dbus/", path)
+
+
+class SpecFileDependencyTestCase(RPMTestCase):
+    """Test UI selection dependency logic using built RPM files."""
+
+    def test_ui_selection_dependencies(self):
+        """Test that UI selection dependencies work correctly in built RPMs."""
+        rpm_paths = self.rpm_paths
+
+        # Test main anaconda package requirements
+        anaconda_rpm = list(filter(RPMFilters.anaconda_main_only, rpm_paths))[0]
```

Comment:
Cosmetic: Note that `filter` and `map` can be replaced with list comprehensions or generator expressions in modern python:
```suggestion
        anaconda_rpm = [p for p in rpm_paths if RPMFilters.anaconda_main_only(p)][0]
```

You could also do it like this:
``` python
        anaconda_rpm = next(p for p in rpm_paths if RPMFilters.anaconda_main_only(p))
```
which will exit after the first truthy value is found instead of filtering and instantiating the whole list.  It's more efficient but that doesn't matter in this code and it may be a bit less readable.